Geometry_Engine/Query/IsContaining.cs

```diff
@@ -174,10 +230,10 @@ public static bool IsContaining(this Polyline curve, List<Point> points, bool ac
                             Vector direction = (end - pPt).Normalise();
                             while (direction.SquareLength() <= 0.5 || edgeDirections.Any(e => 1 - Math.Abs(e.DotProduct(direction)) <= Tolerance.Angle))
```

Comment:
I think it only trigger now if the end and pPt are the same, making the direction into a 0 vector (given it is normalised)

Very likely this could be further refactored, but think this works as is now.

Happy to change if you disagree.